api/src/controllers/cart.js

```diff
@@ -0,0 +1,148 @@
+/* eslint linebreak-style: ["error", "windows"] */
+const {
+  Cart,
+  User,
+  Orderline,
+  Product,
+} = require('../models/index');
+
+const addItem = async (req, res) => {
+  // primero busco el usuario en la base de datos
+  const user = await User.findOne({
+    where: {
+      id: req.body.id,
+    },
+    include: Cart,
+  });
+  // si no lo encuentro mando un error
+  if (user == null) {
+    return res.json({
+      err: 'El usuario no existe',
+    });
+  }
+  // busco si hay un carrito previamente creado del usuario
+  let bandera = false;
+  for (let i = 0; i < user.carts.length; i += 1) {
+    if (user.carts[i].status === 'creada') {
+      bandera = true;
+      break;
+    }
+  }
+  // encuentro el carrito previamente creado
+
+  let cart;
+  if (bandera) {
+    cart = await Cart.findOne({
+      where: {
+        status: 'creada',
+      },
+      include: Orderline,
+    });
+  } else {
+    // SI NO TENE CARRITO PREVIAMENTE CREADO, SE CREA UNO Y SE LO DA AL FRONT
+    cart = await Cart.create({
+
+    });
+  }
+
+  // encuentro los productos enviado desde el front
+  const results = [];
+  for (let i = 0; i < req.body.products.length; i += 1) {
+    results.push(Product.findOne({
+      where: {
+        id: req.body.products[i].id,
+      },
+    }));
+  }
+  // esto lo tuve que hacer porque sino el eslint tira error
+  const arrayProducts = await Promise.all(results);
+  // corroboro que existan y tengan stock disponible, SI NO HAY MANDA ERROR
+  for (let i = 0; i < arrayProducts.length; i += 1) {
+    if (arrayProducts[i] == null) {
```

Comment:
se usa doble igual en lugar de triple igual

Reply:
Expected '===' and instead saw '=='.eslinteqeqeq